pennylane/ops/op_math/change_op_basis.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines a class for compute, uncompute patterns.
+"""
+from collections import Counter
+from functools import reduce
+
+from pennylane import math, queuing
+from pennylane.decomposition import add_decomps, register_resources, resource_rep
+from pennylane.operation import (
+    DiagGatesUndefinedError,
+    EigvalsUndefinedError,
+    MatrixUndefinedError,
+    Operator,
+    SparseMatrixUndefinedError,
+)
+from pennylane.ops.op_math import adjoint
+
+from .composite import CompositeOp, handle_recursion_error
+
+
+def change_op_basis(compute_op, target_op, uncompute_op=None):
+    """Construct an operator which represents the product of the
+    operators provided.
+
+    Args:
+        compute_op (:class:`~.operation.Operator`): A single operator or product that applies quantum operations.
+        target_op (:class:`~.operation.Operator`): A single operator or a product that applies quantum operations.
+        uncompute_op (:class:`~.operation.Operator`): A single operator or a product that applies quantum operations. Default is uncompute_op=qml.adjoint(compute_op).
+
+    Returns:
+        ~ops.op_math.ChangeOpBasis: the operator representing the compute, uncompute pattern.
+    """
+
+    ops_simp = ChangeOpBasis(compute_op, target_op, uncompute_op)
+
+    return ops_simp
```

Comment:
```suggestion
    return ChangeOpBasis(compute_op, target_op, uncompute_op)
```
If all this function does is to pass everything into the constructor, I wonder why we have this function at all.

Reply:
Yeah it is just to be consistent with the UI of `prod`, etc.